cmd/scollector/collectors/extrahop.go

```diff
@@ -0,0 +1,172 @@
+package collectors
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"bosun.org/_third_party/github.com/kylebrandt/gohop"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+const extraHopIntervalSeconds int = 30
+
+var extraHopFilterProtoBy string  //What to filter the traffic by. Valid values are "namedprotocols", "toppercent" or "none"
+var extraHopTopProtoPerc int      //Only log the top % of protocols by volume
+var extraHopOtherProtoName string //What name to log the "other" data under.
+var extraHopL7Description string  //What to append to the end of the L7 description metadata to explain what is and isn't filtered out
+
+//Register a collector for ExtraHop
+func ExtraHop(host, apikey, filterby string, filterpercent int) error {
+	if host == "" || apikey == "" {
+		return fmt.Errorf("Empty host or API key for ExtraHop.")
+	}
+	extraHopFilterProtoBy = filterby
+	switch filterby { //Set up options
+	case "toppercent":
+		extraHopL7Description = fmt.Sprintf("Only the top %d percent of traffic has its protocols logged, the remainder is tagged as as proto=otherprotos", extraHopTopProtoPerc)
+		extraHopOtherProtoName = "otherprotos"
+		if filterpercent > 0 && filterpercent < 100 {
+			extraHopTopProtoPerc = filterpercent
+		} else {
+			return fmt.Errorf("Invalid ExtraHop FilterPercent value (%d). Number should be between 1 and 99.", filterpercent)
+		}
+	case "namedprotocols":
+		extraHopL7Description = "Only named protocols are logged. Any unnamed protocol (A protocol name starting with tcp, udp or ssl) is tagged as proto=unnamed"
+		extraHopOtherProtoName = "unnamed"
+	case "none":
+		extraHopL7Description = ""
```

Comment:
I left them in place deliberately so that it would be obious that we don't want to set them to anything (and not that I forgot to code something for "none"). I can replace it with a comment though.